### PR TITLE
Infineon TC3xx support

### PR DESCRIPTION
**Pull request by bigbrett**
- Adds support for Infineon AURIX TC3xx family (TC375 AURIX LiteKit-V2). See [IDE/AURIX/README.md](https://github.com/bigbrett/wolfBoot/blob/aurix-tc3xx-support/IDE/AURIX/README.md) for details
- Adds non-default option for using a multi-sector erase for the remainder of empty sectors while swapping images (`FLASH_MULTI_SECTOR_ERASE`/`WOLFBOOT_FLASH_MULTI_SECTOR_ERASE`)
- Suppresses a few wolfBoot unused variable compiler warnings

### CODE REVIEW

**Review by dgarske — CHANGES_REQUESTED** [review]

File: IDE/AURIX/README.md
Comment (bigbrett) [reply IDE/AURIX/README.md:5]: @dgarske looks like you are building the TASKING configuration, which isn't supported. See [here ](https://github.com/bigbrett/wolfBoot/blob/ba7cd3d59670008aa95db9bfe5a37a7f73b57da3/IDE/AURIX/README.md?plain=1#L12) and [here](https://github.com/bigbrett/wolfBoot/blob/ba7cd3d59670008aa95db9bfe5a37a7f73b57da3/IDE/AURIX/README.md?plain=1#L73) in the README.

Let me know if that doesn't fix it!

File: .github/workflows/test-powerfail-simulator.yml
Comment (danielinux) [inline .github/workflows/test-powerfail-simulator.yml:584]: Instead of this, add the option to the `make` command line at l:592

File: IDE/AURIX/test-app/Cpu0_Main.c
Comment (bigbrett) [reply IDE/AURIX/test-app/Cpu0_Main.c]: done

File: src/update_flash.c
Comment (bigbrett) [reply src/update_flash.c:667]: done

**Review by dgarske — CHANGES_REQUESTED** [review]
Revised instructions works perfect. I don't have hardware to test, but was able to build and signed. Also please squash this once you resolve the license issues.

File: IDE/AURIX/test-app/Lcf_Gnuc_Tricore_Tc.lsl
Comment (bigbrett) [reply IDE/AURIX/test-app/Lcf_Gnuc_Tricore_Tc.lsl]: done and squashed

File: hal/aurix_tc3xx.c
Comment (dgarske) [inline hal/aurix_tc3xx.c:103]: Mixed declaration. Move variable definitions to top.

File: hal/aurix_tc3xx.c
Comment (dgarske) [inline hal/aurix_tc3xx.c:113]: size_t offset. Mixed declaration. Move variable definitions to top.

File: hal/aurix_tc3xx.c
Comment (dgarske) [inline hal/aurix_tc3xx.c:78]: NIT: `else if () {`

File: hal/aurix_tc3xx.c
Comment (dgarske) [inline hal/aurix_tc3xx.c:168]: Mixed declaration. Move variable definitions to top. Several other places

File: hal/aurix_tc3xx.c
Comment (dgarske) [inline hal/aurix_tc3xx.c:339]: Move below variable declarations.

File: hal/aurix_tc3xx.c
Comment (dgarske) [inline hal/aurix_tc3xx.c:351]: Please don't use bool. Use `int` and 1=true, 0=false.

**Review by dgarske — CHANGES_REQUESTED** [review]

File: hal/aurix_tc3xx.c
Comment (bigbrett) [reply hal/aurix_tc3xx.c:103]: that one is const, so must always will be assigned and declared at the same time

File: hal/aurix_tc3xx.c
Comment (bigbrett) [reply hal/aurix_tc3xx.c:113]: done

File: hal/aurix_tc3xx.c
Comment (bigbrett) [reply hal/aurix_tc3xx.c]: done

File: hal/aurix_tc3xx.c
Comment (bigbrett) [reply hal/aurix_tc3xx.c]: done

File: hal/aurix_tc3xx.c
Comment (bigbrett) [reply hal/aurix_tc3xx.c]: done

File: hal/aurix_tc3xx.c
Comment (dgarske) [inline hal/aurix_tc3xx.c:270]: Mixed declaration.

**Review by dgarske — CHANGES_REQUESTED** [review]
Once more mixed declaration. Otherwise you'll get my approval. Then I'd like @danielinux to get a final say on merge.

File: hal/aurix_tc3xx.c
Comment (bigbrett) [reply hal/aurix_tc3xx.c]: d'oh! fixed.

**Review by dgarske — APPROVED** [review]

**Review by danielinux — APPROVED** [review]
Approved. Thanks for taking the time to update the documentation.

**Pull request merged**
